testing/scenario_app/ios/Scenarios/ScenariosUITests/GoldenImage.m

```diff
@@ -0,0 +1,101 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "GoldenImage.h"
+#import <XCTest/XCTest.h>
+#include <sys/sysctl.h>
+
+@interface GoldenImage ()
+
+@property(readwrite, strong, nonatomic) NSString* goldenName;
+@property(readwrite, strong, nonatomic) UIImage* image;
+@property(strong, nonatomic) NSString* platformName;
+
+@end
+
+@implementation GoldenImage
+
+- (instancetype)initWithGoldenNamePrefix:(NSString*)prefix {
+  self = [super init];
+  if (self) {
+    _goldenName = [prefix stringByAppendingString:self.platformName];
+    NSBundle* bundle = [NSBundle bundleForClass:[self class]];
+    NSURL* goldenURL = [bundle URLForResource:_goldenName withExtension:@"png"];
+    NSData* data = [NSData dataWithContentsOfURL:goldenURL];
+    _image = [[UIImage alloc] initWithData:data];
+  }
+  return self;
+}
+
+- (BOOL)compareGoldenToImage:(UIImage*)image {
```

Comment:
Override `-isEqual`: and `-hash` instead.

Reply:
> Override `-isEqual`: and `-hash` instead.

Oh actually never mind, that would require you to make a GoldenImage out of the screen shot.  This is fine.